command/agent.go

```diff
@@ -238,6 +241,7 @@ func (cmd *AgentCommand) readConfig() *agent.Config {
 	cmdCfg.DNSRecursors = append(cmdCfg.DNSRecursors, dnsRecursors...)
 
 	cfg = agent.MergeConfig(cfg, &cmdCfg)
+	disableHostNodeID.Merge(cfg.DisableHostNodeID)
```

Comment:
If the CLI argument was set (pointer is non-null), then this will take its value and lay it on top of the one from the config files. This ended up being kind of nasty because we had a CLI version of the flag, not just a config one, so it needed this special handling.